pkg/feature/api/flag_trigger.go

```diff
@@ -17,62 +17,987 @@ package api
+	}
+	limit := int(request.PageSize)
+	cursor := request.Cursor
+	if cursor == "" {
+		cursor = "0"
+	}
+	offset, err := strconv.Atoi(cursor)
+	if err != nil {
+		dt, err := statusInvalidCursor.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "cursor"),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	flagTriggers, nextOffset, totalCount, _ := v2fs.NewFlagTriggerStorage(s.mysqlClient).ListFlagTriggers(
+		ctx,
+		whereParts,
+		orders,
+		limit,
+		offset,
+	)
+	triggerWithUrls := make([]*featureproto.ListFlagTriggersResponse_FlagTriggerWithUrl, 0, len(flagTriggers))
+	for _, trigger := range flagTriggers {
+		flagTriggerSecret := domain.NewFlagTriggerSecret(
+			trigger.Id,
+			trigger.FeatureId,
+			trigger.EnvironmentNamespace,
+			trigger.Uuid,
+			int(trigger.Action),
+		)
+		triggerURL, err := s.generateTriggerURL(ctx, flagTriggerSecret, true)
+		if err != nil {
+			s.logger.Error(
+				"Failed to generate trigger url",
+				log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+			)
+			return nil, err
+		}
+		triggerWithUrls = append(triggerWithUrls, &featureproto.ListFlagTriggersResponse_FlagTriggerWithUrl{
+			FlagTrigger: trigger,
+			Url:         triggerURL,
+		})
+	}
+	return &featureproto.ListFlagTriggersResponse{
+		FlagTriggers: triggerWithUrls,
+		Cursor:       strconv.Itoa(nextOffset),
+		TotalCount:   totalCount,
+	}, nil
+}
+
+func (s *FeatureService) newListFlagTriggerOrders(
+	orderBy featureproto.ListFlagTriggersRequest_OrderBy,
+	orderDirection featureproto.ListFlagTriggersRequest_OrderDirection,
+	localizer locale.Localizer,
+) ([]*mysql.Order, error) {
+	var column string
+	switch orderBy {
+	case featureproto.ListFlagTriggersRequest_DEFAULT, featureproto.ListFlagTriggersRequest_CREATED_AT:
+		column = "created_at"
+	case featureproto.ListFlagTriggersRequest_UPDATED_AT:
+		column = "updated_at"
+	default:
+		dt, err := statusInvalidOrderBy.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "order_by"),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	direction := mysql.OrderDirectionAsc
+	if orderDirection == featureproto.ListFlagTriggersRequest_DESC {
+		direction = mysql.OrderDirectionDesc
+	}
+	return []*mysql.Order{
+		mysql.NewOrder(column, direction),
+	}, nil
+}
+
+func (s *FeatureService) FlagTriggerWebhook(
+	ctx context.Context,
+	request *featureproto.FlagTriggerWebhookRequest,
+) (*featureproto.FlagTriggerWebhookResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+	secret := request.GetSecret()
+	resp := &featureproto.FlagTriggerWebhookResponse{}
+	if secret == "" {
+		s.logger.Error(
+			"Failed to get secret from query",
+			log.FieldsFromImcomingContext(ctx)...,
+		)
+		return resp, nil
```

Comment:
If there is an error, we must return the error so the caller can know what happened.

Reply:
Also, please add unit tests to cover those cases.

Reply:
OK.